### DISCUSSION TITLE
Need clarification on "ID – Procurement documents" ("OPT-140-Lot")

### DISCUSSION THREAD

**Discussion by VR-M** [OP-TED/eForms-SDK#863] [Q&A] [answered]
<cac:CallForTendersDocumentReference> 
    **<cbc:ID></cbc:ID>**
   <cbc:DocumentType>non-restricted-document</cbc:DocumentType>
   <cac:Attachment>
      <cac:ExternalReference>                       
          <cbc:URI>https://xxxxxxxxxxxxx</cbc:URI>
      </cac:ExternalReference>
   </cac:Attachment>
</cac:CallForTendersDocumentReference> 

You see the ID above which is also the "OPT-140-Lot" is empty. This is one of the example for Contract Notice that we tested. We have a strange case where:
1. If the Id is added to the xml whether it has data or not, the cvs validation passed.
2. If I remove the ID from the xml (for cases of of no data like the example above), the CVS validation failed.
It does not make sense to add the ID if there is no value in it.

Do you observe this and is it correct at all to add element with no value to the xml?

**Comment by YvesJo** [comment, marked as answer]
Hi,
For about 2 years, there is a stage 3 rule that checks that elements aren't empty. Could you may be provide further information (SDK version, XML instance ...) on the submitted cases where cbc:ID would be empty and pass the CVS validation?
cbc:ID is mandatory following the XSD, this is required by the UBL standard. First value is satisfy the schema, second value could be explicitly specify a relevant identifier for the documents.
KR

**Comment by VR-M** [comment]
Hi,
The sdk version is 1.9 and the XML instance has been attached.
We tested the XML via Swagger UI.
[testInstance.xml.zip](https://github.com/OP-TED/eForms-SDK/files/14649083/testInstance.xml.zip)

**Reply by YvesJo** [reply]
Hi,
Doing the test, we get the following report: [report(3).zip](https://github.com/OP-TED/eForms-SDK/files/14649481/report.3.zip)
with the error on lines 121 to 125 for the empty element

**Comment by VR-M** [comment]
Hi,
Thanks for the response. 
I tested the similar notice in TED enotices2 preview portal. I did not fill the ID-Procurement and validated. It gave no error.
I have
![Uploading Screenshot 2024-03-19 at 5.05.23 PM.png…]()
 attached the screenshot for reference.

**Reply by YvesJo** [reply]
Hi,
there is no attached screenshot.
In absence of specified value, when OPT-140 has to exist, eNotices2 is adding a default value (DEFAULT_VALUE_CHANGE_ME) that you should be able to find in the XML when you export that file for the notice. This is why this rule is not triggered for enotices2 users.
